apps/mobile/src/app/account/[accountId]/activity.tsx

```diff
@@ -1,25 +1,57 @@
 import { configureAccountParamsSchema } from '@/app/account/[accountId]/index';
 import { FetchWrapper } from '@/components/loading';
-import { ActivityEmpty, ActivityLayout, ActivityList, hasActivity } from '@/features/activity';
+import { Screen } from '@/components/screen/screen';
+import { ActivityEmpty, ActivityListItem } from '@/features/activity';
+import { useRefreshHandler } from '@/features/refresh-control/refresh-control';
 import { useAccountActivity } from '@/queries/activity/account-activity.query';
 import { deserializeAccountId } from '@/store/accounts/accounts';
+import { i18n } from '@lingui/core';
+import { t } from '@lingui/macro';
 import { useLocalSearchParams } from 'expo-router';
 
+import { OnChainActivity } from '@leather.io/models';
+import { Text } from '@leather.io/ui/native';
+
 export default function AccountActivityScreen() {
   const params = useLocalSearchParams();
   const { accountId, accountName } = configureAccountParamsSchema.parse(params);
+  const { refreshing, onRefresh } = useRefreshHandler();
   const { fingerprint, accountIndex } = deserializeAccountId(accountId);
   const activity = useAccountActivity(fingerprint, accountIndex);
 
   return (
-    <ActivityLayout accountName={accountName}>
+    <Screen>
+      <Screen.Header
+        centerElement={
+          <Text variant="label01">
+            {t({
+              id: 'activity.account.header_title',
+              message: 'Activity',
+            })}
+          </Text>
+        }
+      />
       <FetchWrapper data={activity}>
-        {hasActivity(activity) ? (
-          activity.state === 'success' && <ActivityList activity={activity.value} />
-        ) : (
-          <ActivityEmpty />
+        {activity.state === 'success' && (
+          <Screen.List
+            ListHeaderComponent={
+              <Screen.Title>
+                {i18n._({
+                  id: 'activity.account.header_content_title',
+                  message: '{accountName} Activity',
+                  values: { accountName: accountName },
+                })}
+              </Screen.Title>
+            }
+            data={activity.value as OnChainActivity[]} // TODO: Unclear why was this cast. Needs clearing up.
```

Comment:
Without casting this you get other type errors like this:
```
Type 'Activity' is not assignable to type 'OnChainActivity'.
  Type 'ConnectAppActivity' is not assignable to type 'OnChainActivity'.
    Type 'ConnectAppActivity' is missing the following properties from type 'SwapAssetsActivity': fromAsset, fromAmount, toAsset, toAmount, and 2 more.
```

Re-visiting this and I think it was a mistake on my part not always specifying this type of activity data as `OnChainActivity` / I struggled to properly narrow it in all cases. 

It would be great to clean this up as the activity list only shows `OnChainActivity`